test/test_tensor_creation_ops.py

```diff
@@ -832,6 +832,139 @@ def compare_helper_(like_fn, t):
+                if shape[dim] % arg == 0:
+                    result = torch.hsplit(t, arg)
+                    result_n = np.hsplit(t_n, arg)
+                    self.assertEqual(result, result_n)
+                else:
+                    with self.assertRaisesRegex(RuntimeError, f"torch.hsplit's input axis : {dim} has dimension size : "
+                                                              + f"{ shape[dim]} which is not divisible by split_size"
+                                                              + f"={arg}"):
+                        torch.hsplit(t, arg)
+                    with self.assertRaises(ValueError):
+                        np.hsplit(t_n, arg)
+
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
+    def test_vsplit(self, device, dtype):
+        inputs = (
+            ((6,), 2),
+            ((6,), 4),
+            ((6, 5), 2),
+            ((6, 5), 4),
+            ((6, 5), [1, 2, 3]),
+            ((6, 5), [1, 5, 9]),
+            ((6, 5, 5), 2),
+            ((6, 0, 5), 2),
+            ((5, 0, 5), [1, 5]),
+        )
+
+        for shape, arg in inputs:
+            t = make_tensor(shape, device, dtype)
+            t_n = t.cpu().numpy()
+
+            if len(shape) < 2:
+                with self.assertRaisesRegex(RuntimeError, "torch.vsplit requires a tensor with at least 2 dimension"):
+                    torch.vsplit(t, arg)
+                with self.assertRaises(ValueError):
+                    np.vsplit(t_n, arg)
+
+            elif isinstance(arg, list) or len(shape) == 0:
+                result = torch.vsplit(t, arg)
+                result_n = np.vsplit(t_n, arg)
+                self.assertEqual(result, result_n)
+            else:
+                if shape[0] % arg == 0:
+                    result = torch.vsplit(t, arg)
+                    result_n = np.vsplit(t_n, arg)
+                    self.assertEqual(result, result_n)
+                else:
+                    with self.assertRaisesRegex(RuntimeError, "torch.vsplit's input axis : 0 has dimension size : "
+                                                              + f"{ shape[0] } which is not divisible by split_size"
+                                                              + f"={arg}"):
+                        torch.vsplit(t, arg)
+                    with self.assertRaises(ValueError):
+                        np.vsplit(t_n, arg)
+
+    @onlyOnCPUAndCUDA
+    @dtypes(*torch.testing.get_all_dtypes(include_bfloat16=False))
+    def test_dsplit(self, device, dtype):
+        inputs = (
+            ((6,), 4),
+            ((6, 6), 3),
+            ((5, 5, 6), 2),
+            ((5, 5, 6), 4),
+            ((5, 5, 6), [1, 2, 3]),
+            ((5, 5, 6), [1, 5, 9]),
+            ((5, 5, 0), 2),
+            ((5, 0, 6), 4),
+            ((5, 0, 6), [1, 2, 3]),
+            ((5, 5, 6), [1, 5, 9]),
+        )
+
+        for shape, arg in inputs:
```

Comment:
This tests are good. I challenge you to reimplement these tests using a helper. For the {h, v, d}split I think the "inputs" (AKA the test cases) and the split dimension are different, so the helper can accept the test cases and the split dimension and perform the logic where:

- checks for an error if the tensor has too few dimensions
- checks for an error if the split dimension is not divisible by a scalar split
- checks for consistency with NumPy

In fact, these tests can probably simplify themselves further by running the torch version and the NumPy version and recording exceptions, then verifying that either both threw exceptions or they produced the same output. The torch error message could be evaluated independently afterwards to be sure the language was correct.